File: Divyansh Singh/Array/group anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>>ans;
        
        map<string , vector<string>>mpp;
        
        
        for(int i=0;i<strs.size();i++){
            
            string s=strs[i];
            sort(s.begin(),s.end());
            if (mpp.find(s)!=mpp.end()){
                vector<string>p;
                p=mpp[s];
                p.push_back(strs[i]);
                mpp[s]=p;
                
            }
            
            
            else{
                
                vector<string>p;
                p.push_back(strs[i]);
                mpp[s]=p;
                
            }
        }
        
        
        for(auto itr:mpp){
            ans.push_back(itr.second);
        }
        
        return ans;
    }
};

File: Deepak Urs/Arrays/groupAnagrams.java
import java.util.*;

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> res = new ArrayList<>();
        Map<String, List<String>> mp = new HashMap<>();

        for(int i = 0; i< strs.length; i++) {
            char[] k = strs[i].toCharArray();
            Arrays.sort(k);
            String key = new String(k);

            if(mp.containsKey(key)) {
                mp.get(key).add(strs[i]);
            }
            else {
                List<String> lt = new ArrayList<>();
                lt.add(strs[i]);
                mp.put(key, lt);
            }
        }

        for(String k: mp.keySet()){
            res.add(mp.get(k));
        }

        return res;
    }
}